tests/pfcwd/test_pfc_config.py

```diff
@@ -0,0 +1,257 @@
+        write_file = key
+        write_params = dict()
+        write_params["PFC_WD"] = { test_port: { "action": cfg_params[key]["pfc_wd_action"],
+                                                "detection_time": cfg_params[key]["pfc_wd_detection_time"],
+                                                "restoration_time": cfg_params[key]["pfc_wd_restoration_time"]
+                                              }
+                                  }
+        # create individual template files for each test
+        with open(os.path.join(TMP_DIR, "{}.json".format(write_file)), "w") as wfile:
+            json.dump(write_params, wfile)
+
+    return cfg_params
+
+def copy_templates_to_dut(duthost, cfg_params):
+    """
+    Copy all the templates created to the DUT
+
+    Args:
+        duthost (AnsibleHost): instance
+        cfg_params (dict): all config templates
+
+    Returns:
+        None
+    """
+    duthost.shell("mkdir -p {}".format(DUT_RUN_DIR))
+    for key in cfg_params:
+        src_file = os.path.join(TMP_DIR, "{}.json".format(key))
+        duthost.copy(src=src_file, dest="{}/{}.json".format(DUT_RUN_DIR, key))
+
+def cfg_teardown(duthost):
+    """
+    Cleans up the DUT temp dir and temp dir on the host after the module run
+
+    Args:
+        duthost (AnsibleHost): instance
+
+    Returns:
+        None
+    """
+    if os.path.exists(TMP_DIR):
+        os.system("rm -rf {}".format(TMP_DIR))
+    duthost.shell("rm -rf {}".format(DUT_RUN_DIR))
+
+@pytest.fixture(scope='class', autouse=True)
+def cfg_setup(setup_pfc_test, duthost):
+    """
+    Class level automatic fixture. Prior to the test run, create all the templates
+    needed for each individual test and copy them on the DUT.
+    After the all the test cases are done, clean up temp dir on DUT and host
+
+    Args:
+        setup_pfc_test: module fixture defined in module conftest.py
+        duthost: instance of AnsibleHost class
+    """
+    setup_info = setup_pfc_test
+    pfc_wd_test_port = setup_info['test_ports'].keys()[0]
+    logger.info("Creating json templates for all config tests")
+    cfg_params = generate_cfg_templates(pfc_wd_test_port)
+    logger.info("Copying templates over to the DUT")
+    copy_templates_to_dut(duthost, cfg_params)
+
+    yield
+    logger.info("--- Start running config tests ---")
+
+    logger.info("--- Clean up config dir from DUT ---")
+    cfg_teardown(duthost)
+
+
+@pytest.fixture(scope='function', autouse=True)
+def stop_pfcwd(duthost):
+    """
+    Fixture that stops PFC Watchdog before each test run
+
+    Args:
+        duthost: instance of AnsibleHost class
+
+    Returns:
+        None
+    """
+    logger.info("--- Stop Pfcwd --")
+    duthost.command("pfcwd stop")
+
+
+@pytest.mark.usefixtures('cfg_setup')
+class TestPfcConfig(object):
+    """
+    Test case definition and helper function class
+    """
+    def execute_test(self, duthost, marker, ignore_regex=None, expect_regex=None, expect_errors=False):
```

Comment:
suggest rename marker to syslog_marker for clarity.

Reply:
renamed to 'syslog_marker'